nameresolution/kubernetes/kubernetes.go

```diff
@@ -28,12 +29,19 @@ func NewResolver(logger logger.Logger) nameresolution.Resolver {
 
 // Init initializes Kubernetes name resolver.
 func (k *resolver) Init(metadata nameresolution.Metadata) error {
+	if config, ok := metadata.Configuration.(map[string]string); ok {
+		clusterDomain := config[ClusterDomainKey]
```

Comment:
If you set `k.clusterDomain = DefaultClusterDomain` here, then you don'y need the `if clusterDomain == "" {` in `ResolveID` and makes the code a little easier to follow (not that this change is complicated to begin with)

Reply:
good idea, why I want check `if clusterDomain == ""` in `ResolveID` again, is because I'm afraid user may omit call `resolver.Init`, like the previous test case  https://github.com/dapr/components-contrib/blob/b404517388bd4b036f693b96dd73d1b0a1aff642/nameresolution/kubernetes/kubernetes_test.go#L16

this is what I did now:
1.  `k.clusterDomain = DefaultClusterDomain`  when NewResolver()
2.  remove `if clusterDomain == "" {` in `ResolveID`

please check again: https://github.com/dapr/components-contrib/pull/793/commits/8e42c2fca3085731081800a14a6cad63fdb898d7